java/code/src/com/redhat/rhn/manager/channel/test/BaseRepoCommandTest.java

```diff
@@ -130,4 +130,82 @@ private void validUrlInput(String url, String type) {
             fail("valid repo type caused error: " + url);
         }
     }
+
+    public void testRepoLabelInput() {
+
+        // Repository label must contain only letters, hyphens, periods, underscores and numeral
+
+        // I N V A L I D
+        invalidRepoLabelInput("");
+        invalidRepoLabelInput("example/repo");
+        invalidRepoLabelInput("example;repo");
+        invalidRepoLabelInput("example_repo$");
+        invalidRepoLabelInput("my*examplerepo");
+        invalidRepoLabelInput("example^repo");
+
+        // V A L I D
+        validRepoLabelInput("my example repo");
+        validRepoLabelInput("my-example-repo");
+        validRepoLabelInput("my-example_repo");
+        validRepoLabelInput("my_repo_22");
+        validRepoLabelInput("My-Example-Repo");
+        validRepoLabelInput("My/Example/Repo/15");
```

Comment:
Why is this valid?

Reply:
I used here the same regex we have for validating the channel names. The slashes will be replaced as non-valid characters in the reposync plugins (as for already created custom repository labels in the database).

On the other hand, we could also not accept slashes from the UI to be safer and prevent posible related issues with path maybe even in some other places from the code base where we use the "repository label" to build paths in the system.

Reply:
What I mean is: in Line 140 we have this test:
```
    invalidRepoLabelInput("example/repo");
```

Why is "example/repo" INVALID but "My/Example/Repo/15" is VALID ?
Is it not because of the slash?

Reply:
Oh! That's correct.. there should be a problem with the logic of the `invalidRepoLabelInput` and `validRepoLabelInput` helpers. I'll fix that.

With the current implementation, the "slashes" are allowed.